File: generate/client/dart/lib/model/create_or_update_todolist.dart
part of swagger.api;

class CreateOrUpdateTodolist {
  
  String name = null;

  double priority = null;

  List<String> tags = [];

  CreateOrUpdateTodolist();

  @override
  String toString() {
    return 'CreateOrUpdateTodolist[name=$name, priority=$priority, tags=$tags, ]';
  }

  CreateOrUpdateTodolist.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    name = json['name'];
    priority = json['priority'];
    tags = (json['tags'] as List).map((item) => item as String).toList();
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'priority': priority,
      'tags': tags
     };
  }

  static List<CreateOrUpdateTodolist> listFromJson(List<dynamic> json) {
    return json == null ? new List<CreateOrUpdateTodolist>() : json.map((value) => new CreateOrUpdateTodolist.fromJson(value)).toList();
  }

  static Map<String, CreateOrUpdateTodolist> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, CreateOrUpdateTodolist>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new CreateOrUpdateTodolist.fromJson(value));
    }
    return map;
  }
}

File: generate/server/java/src/main/java/io/swagger/server/api/model/CreateOrUpdateTodolist.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * CreateOrUpdateTodolist
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaVertXServerCodegen", date = "2023-10-11T18:50:12.795023015Z[GMT]")



public class CreateOrUpdateTodolist   {
  private String name = null;

  private BigDecimal priority = null;

  
  private List<String> tags = null;

  public CreateOrUpdateTodolist name(String name) {
    this.name = name;
    return this;
  }

  /**
   * Get name
   * @return name
   **/
    public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public CreateOrUpdateTodolist priority(BigDecimal priority) {
    this.priority = priority;
    return this;
  }

  /**
   * Get priority
   * @return priority
   **/
    public BigDecimal getPriority() {
    return priority;
  }

  public void setPriority(BigDecimal priority) {
    this.priority = priority;
  }

  public CreateOrUpdateTodolist tags(List<String> tags) {
    this.tags = tags;
    return this;
  }

  public CreateOrUpdateTodolist addTagsItem(String tagsItem) {
    if (this.tags == null) {
      this.tags = new ArrayList<String>();
    }
    this.tags.add(tagsItem);
    return this;
  }

  /**
   * Get tags
   * @return tags
   **/
    public List<String> getTags() {
    return tags;
  }

  public void setTags(List<String> tags) {
    this.tags = tags;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    CreateOrUpdateTodolist createOrUpdateTodolist = (CreateOrUpdateTodolist) o;
    return Objects.equals(this.name, createOrUpdateTodolist.name) &&
        Objects.equals(this.priority, createOrUpdateTodolist.priority) &&
        Objects.equals(this.tags, createOrUpdateTodolist.tags);
  }

  @Override
  public int hashCode() {
    return Objects.hash(name, priority, tags);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class CreateOrUpdateTodolist {\n");
    
    sb.append("    name: ").append(toIndentedString(name)).append("\n");
    sb.append("    priority: ").append(toIndentedString(priority)).append("\n");
    sb.append("    tags: ").append(toIndentedString(tags)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
